packages/schemas/src/gen/schema.ts

```diff
@@ -33,25 +33,31 @@ export const generateSchema = ({ name, fields }: TableWithType) => {
     '};',
     '',
     `const createGuard: CreateGuard<${databaseEntryType}> = z.object({`,
-    // eslint-disable-next-line complexity
-    ...fields.map(({ name, type, isArray, isEnum, nullable, hasDefaultValue, tsType }) => {
-      if (tsType) {
-        return `  ${camelcase(name)}: ${camelcase(tsType)}Guard${conditionalString(
-          nullable && '.nullable()'
-        )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
-      }
 
-      return `  ${camelcase(name)}: z.${
-        isEnum ? `nativeEnum(${type})` : `${type}()`
-      }${conditionalString(isArray && '.array()')}${conditionalString(
-        nullable && '.nullable()'
-      )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
-    }),
+    ...fields.map(
+      // eslint-disable-next-line complexity
+      ({ name, type, isArray, isEnum, nullable, hasDefaultValue, tsType, isString, maxLength }) => {
+        if (tsType) {
+          return `  ${camelcase(name)}: ${camelcase(tsType)}Guard${conditionalString(
+            nullable && '.nullable()'
+          )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
+        }
+
+        return `  ${camelcase(name)}: z.${
+          isEnum ? `nativeEnum(${type})` : `${type}()`
+        }${conditionalString(isString && maxLength && `.max(${maxLength})`)}${conditionalString(
+          isArray && '.array()'
+        )}${conditionalString(nullable && '.nullable()')}${conditionalString(
+          (nullable || hasDefaultValue) && '.optional()'
+        )},`;
+      }
```

Comment:
nit: would it be convenient to have a type parser:

```js

const renderPrimaryType = (type: string, isEnum: boolean) => {
  if (isEnum ) return `z.nativeEnum(${type})`;
  return `z.${type}()`
};

const renderRules = ({isArray, maxStringLength }) => {
  return ${conditionalString(maxStringLength && `.max(${maxStringLength})`)}${conditionalString(
          isArray && '.array()'
    )}
};

const renderNullishOptions = (nullable: Boolean, optional: Boolean) => {
 return ${conditionalString(nullable && '.nullable()')}${conditionalString((nullable || hasDefaultValue) && '.optional()';
};


const parseZodType = ({type: PrimaryType, stringMaxLength: number, isEnum: Boolean,  isArray: Boolean nullable: Boolean, optional: Boolean } ) => {
  return `${renderPrimaryType(type, isEnum)}${renderRules(stringMaxLength, isArray)}${renderNullishOptions(nullable, optioanl)}`;
}
```


In the future we could update the field type schema better

Reply:
We can encapsulate the zod guard code builder in `ColumnDefinition` parse from table.sql in the future.